engine/grid.py

```diff
@@ -600,6 +699,58 @@ def add_guided_waypoints(self, is_vertical):
             else:
                 self.curr_pos = (turning_column, next_row)
 
+    def simple_get_guided_waypoints(self):
+        flip = True
+        if (
+            self.direction == self.direction.UP
+            or self.direction == self.direction.RIGHT
+        ):
+            flip = False
+        if self.direction == self.direction.UP or self.direction == self.direction.DOWN:
+            for row in range(len(self.vertical_waypoint_list)):
+                # if self.waypoints:
+                #     curr_point = self.waypoints[-1]
+                #     next_point = self.vertical_waypoint_list[row][0]
+                # TODO add turning function that returns list of Nodes
+                # circle_plt = turning_function(curr_point, next_point)
+                # self.waypoints += circle_plt
+                if len(self.vertical_waypoint_list[row]) == 0:
+                    continue
+                if flip:
+                    self.waypoints += [
+                        self.vertical_waypoint_list[row][-1],
+                        self.vertical_waypoint_list[row][0],
+                    ]
+                else:
+                    self.waypoints += [
+                        self.vertical_waypoint_list[row][0],
+                        self.vertical_waypoint_list[row][-1],
+                    ]
```

Comment:
funny way to simplify:
```
tmp = self.vertical_waypoint_list[row] 
self.waypoints += [tmp[-flip], tmp[-(not flip)]]
```
not sure if its a good change

Reply:
Smart! not as readable though... I think we can afford to keep it without the optimization